Source/FieldSolver/FiniteDifferenceSolver/HybridPICModel/HybridPICModel.cpp

```diff
@@ -114,6 +135,16 @@ void HybridPICModel::AllocateLevelMFs (ablastr::fields::MultiFabRegister & field
         lev, amrex::convert(ba, jz_nodal_flag),
         dm, ncomps, IntVect(1), 0.0_rt);
 
+    if (m_add_external_fields) {
+        m_external_vector_potential->AllocateLevelMFs(
+            fields,
+            lev, ba, dm,
+            ncomps, ngEB,
+            Ex_nodal_flag, Ey_nodal_flag, Ez_nodal_flag,
+            Bx_nodal_flag, By_nodal_flag, Bz_nodal_flag
```

Comment:
I think the idea is to move away from passing these arguments around and I believe there is a method now that will create a new multifab using an existing one as template, such that in `AllocateLevelMFs` you would grab E from the field register, and then just make the new multifab you want off it. I'll search for that function tomorrow or we can ask on Slack.

Reply:
This is following what the HybridPICModel is doing. Can we do this as a follow up clean up PR for both objects?